spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/ColumnarBatchReader.java

```diff
@@ -45,11 +45,23 @@ public class ColumnarBatchReader extends BaseBatchReader<ColumnarBatch> {
   private final boolean hasIsDeletedColumn;
   private DeleteFilter<InternalRow> deletes = null;
   private long rowStartPosInBatch = 0;
+  // In the case of Equality Delete, we have also built ColumnarBatchReader for the equality delete
+  // filter columns to read the value to find out which rows are deleted. If these deleted filter
+  // columns are not in the requested schema, then these are the extra columns that we want to
+  // remove before return the ColumnBatch to Spark.
+  // Supposed table schema is C1, C2, C3, C4, C5, The query is:
+  // SELECT C5 FROM table, and the equality delete Filter is on C3, C4,
+  // We read the values of C3, C4 to figure out which rows are deleted, but we don't want to include
+  // these values in the ColumnBatch that we return to Spark. In this example, the numOfExtraColumns
+  // is 2. Since when creating the DeleteFilter, we append these extra columns in the end of the
+  // requested schema, we can just remove them from the end of the ColumnVector.
```

Comment:
[doubt] Is it possible to fix this at the place where these extra columns are appended in the end of the requested schema, this would probably help us in avoiding extra memory in the first place and expensive copy of command ?

Reply:
Thanks for your comment!

 The extra columns are appended to the requested schema in DeleteFilter.[fileProjection](https://github.com/apache/iceberg/blob/06dc721498d6ad95c86f0f884b8ad30f807ef321/data/src/main/java/org/apache/iceberg/data/DeleteFilter.java#L306). The values of these extra columns are read in ColumnarBatchReader and used to identify which rows are deleted in [applyEqDelete.](https://github.com/apache/iceberg/blob/06dc721498d6ad95c86f0f884b8ad30f807ef321/spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/data/vectorized/ColumnarBatchReader.java#L244) I remove the extra columns right after calling applyEqDelete.

Reply:
Thank you for the response !

Considering applyEquality delete anyways does another projection on top of the schema returned from DeleteFilte.[fileProjection](https://github.com/apache/iceberg/blob/06dc721498d6ad95c86f0f884b8ad30f807ef321/data/src/main/java/org/apache/iceberg/data/DeleteFilter.java#L306)

https://github.com/apache/iceberg/blob/bf8d25fe1578ef199d64fb609c0299728ec58910/data/src/main/java/org/apache/iceberg/data/DeleteFilter.java#L196

can we not add one another param in the fileProjection like we did here to include additional fiels based on the boolean flag : 
https://github.com/apache/iceberg/blob/06dc721498d6ad95c86f0f884b8ad30f807ef321/data/src/main/java/org/apache/iceberg/data/DeleteFilter.java#L272